core/src/block.rs

```diff
@@ -363,30 +361,50 @@ pub struct ValidBlock {
     pub rejected_transactions: Vec<VersionedRejectedTransaction>,
     /// Array of all transactions in this block.
     pub transactions: Vec<VersionedValidTransaction>,
-    /// Signatures of peers which approved this block.
-    pub signatures: BTreeSet<SignatureOf<Self>>,
     /// Event recommendations.
     pub event_recommendations: Vec<Event>,
 }
 
 impl ValidBlock {
+    /// Calculate hash of the current block.
+    pub fn hash(&self) -> HashOf<Self> {
```

Comment:
`#[inline]`. The hashing function isn't inlineable, but the transmute is, so you remove the overhead of one extra function call.